src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -387,7 +387,9 @@ void InstanceKlass::set_nest_host(InstanceKlass* host) {
   _nest_host = host;
   // Record dependency to keep nest host from being unloaded before this class.
   ClassLoaderData* this_key = class_loader_data();
-  this_key->record_dependency(host);
+  if (this_key != NULL) {
```

Comment:
I did see this one trigger, otherwise I wouldn't have known about it, but I can't reproduce it today. Whether it's an assert or a guarantee depends on how serious the problem would be.

Reply:
Interesting! I do hit the assert during JDK build:
```
#  Internal Error (.../src/hotspot/share/oops/instanceKlass.cpp:390), pid=956, tid=6147
#  Error: assert(this_key != __null) failed

V   report_vm_error(char const*, int, char const*, char const*, ...)+0x88
V   InstanceKlass::set_nest_host(InstanceKlass*)+0x254
V   SystemDictionary::load_shared_lambda_proxy_class(InstanceKlass*, Handle, Handle, PackageEntry*, JavaThread*)+0x19c
V   SystemDictionaryShared::prepare_shared_lambda_proxy_class(InstanceKlass*, InstanceKlass*, JavaThread*)+0x13c
V   JVM_LookupLambdaProxyClassFromArchive+0x2cc
C  Java_java_lang_invoke_LambdaProxyClassArchive_findFromArchive+0x4c
j  java.lang.invoke.LambdaProxyClassArchive.findFromArchive(...) java.base@20-internal
...
```

Looks like a pre-existing bug to me.

Reply:
OK! I'll do a bit more digging.

Reply:
FTR Calvin filed [JDK-8296433](https://bugs.openjdk.org/browse/JDK-8296433) to track the issue.